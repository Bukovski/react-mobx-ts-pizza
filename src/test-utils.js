import React from 'react';
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import { Provider } from "mobx-react";
import { render } from "@testing-library/react";
import { rootStore } from "./store";


export const renderWithRouter = (component) => {
	const history = createMemoryHistory();
	
	return {
		...render (
			<Router history={ history }>
				{ component }
			</Router>
		)
	}
}

export const renderWithRouterMobx = (component,
                                     { initialState } = {},
                                     store = {
	...rootStore, ...initialState }) => {
	const history = createMemoryHistory()
	
	return {
		...render(
			<Router history={ history }>
				<Provider { ...store }>
					{ component }
				</Provider>
			</Router>
		),
		store
	}
}
