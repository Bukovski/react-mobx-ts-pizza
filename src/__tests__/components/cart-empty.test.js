import React from 'react';
import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../test-utils";
import CartEmpty from "../../components/cart-empty.component";


describe("CartEmpty Component", () => {
	it("should render with Link", () => {
		const { container } = renderWithRouter(<CartEmpty />);
		
		expect(screen.getByText(/Корзина пуста/)).toBeInTheDocument();
		expect(screen.getByText('Вернуться назад').closest('a')).toHaveAttribute('href', '/')
		
		expect(container.firstChild).toMatchSnapshot()
	})
});
