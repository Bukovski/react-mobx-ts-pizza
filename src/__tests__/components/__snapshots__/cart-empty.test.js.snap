// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartEmpty Component should render with Link 1`] = `
<div
  class="cart cart--empty"
>
  <h2>
    –ö–æ—Ä–∑–∏–Ω–∞ –ø—É—Å—Ç–∞ 
    <i>
      üòï
    </i>
  </h2>
  <p>
    –í–µ—Ä–æ—è—Ç–Ω–µ–π –≤—Å–µ–≥–æ, –≤—ã –Ω–µ –∑–∞–∫–∞–∑—ã–≤–∞–ª–∏ –µ—â—ë –ø–∏—Ü—Ü—É.
    <br />
    –î–ª—è —Ç–æ–≥–æ, —á—Ç–æ–±—ã –∑–∞–∫–∞–∑–∞—Ç—å –ø–∏—Ü—Ü—É, –ø–µ—Ä–µ–π–¥–∏ –Ω–∞ –≥–ª–∞–≤–Ω—É—é —Å—Ç—Ä–∞–Ω–∏—Ü—É.
  </p>
  <img
    alt="Empty cart"
    src="empty-cart.png"
  />
  <a
    class="button button--black"
    href="/"
  >
    <span>
      –í–µ—Ä–Ω—É—Ç—å—Å—è –Ω–∞–∑–∞–¥
    </span>
  </a>
</div>
`;
